fda/grid.py

```diff
@@ -700,22 +706,24 @@ def set_figure_and_axes(self):
         else:
             projection = '3d'
 
-        ncols = math.ceil(math.sqrt(self.ndim_image))
-        nrows = math.ceil(self.ndim_image / ncols)
+        if ncols is None and nrows is None:
```

Comment:
If only one of those is `None`, the other can be inferred.

Reply:
changed

Reply:
changed